nodeJs/processors/processMessage.js

```diff
@@ -2,15 +2,20 @@ import { mkdir, writeFile, stat } from "fs";
 import { formatDateForFileName } from "../helpers/dateHelpers.js";
 import path from "path";
 
-const DATA_DIRECTORY = "data";
+function isEmptyOrSpaces(string){
```

Comment:
The `isNullOrWhitespace` method in C# is handy, but I think for our case in Javascript just checking that the `downloadDirectory` is truthy will suffice - i.e. checking by `if(!downloadDirectory)` {... }`

Reply:
Changing to the suggestion will return true even if a user enters space or tab characters eg. has `downloadDirectory = "  "`. Are we OK to allow users to specify a blank string?

Reply:
I personally think so - they will have had to change the default to that. I'm happy to check for that, though, if you prefer. If we do we should change this to an arrow function to be consistent with the style of the repository.

Reply:
I've refactored the error messages for missing config values across the clients in the latest commit. For now I'm keeping the space check for consistency with the C# client, although happy for this to be reconsidered in a future PR that focuses on improving and standardising error messages across the clients.